In [7]:
import pandas as pd

In [17]:
df = pd.read_csv("../data/train.csv", index_col="PassengerId") 
df.head()

Survived  Pclass  \
PassengerId                     
1                   0       3   
2                   1       1   
3                   1       3   
4                   1       1   
5                   0       3   

                                                          Name     Sex   Age  \
PassengerId                                                                    
1                                      Braund, Mr. Owen Harris    male  22.0   
2            Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0   
3                                       Heikkinen, Miss. Laina  female  26.0   
4                 Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0   
5                                     Allen, Mr. William Henry    male  35.0   

             SibSp  Parch            Ticket     Fare Cabin Embarked  
PassengerId                                                          
1                1      0         A/5 21171   7.2500   NaN        S  
2                1      0          PC 17599  71.2833   C85        C  
3                0      0  STON/O2. 3101282   7.9250   NaN        S  
4                1      0            113803  53.1000  C123        S  
5                0      0            373450   8.0500   NaN        S

In [67]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

df=sns.load_dataset('titanic')
df.head()

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   
2         1       3  female  26.0      0      0   7.9250        S  Third   
3         1       1  female  35.0      1      0  53.1000        S  First   
4         0       3    male  35.0      0      0   8.0500        S  Third   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False  
2  woman       False  NaN  Southampton   yes   True  
3  woman       False    C  Southampton   yes  False  
4    man        True  NaN  Southampton    no   True

In [43]:
df[['sex','survived']].groupby('sex', as_index=False).mean()

sex  survived
0  female  0.742038
1    male  0.188908

여성 생존율과 남성 생존율 비교해보니 여성의 생존율이 남성의 생존율보다 매우 높은 것으로 보인다.

In [95]:
import pandas as pd

# 데이터셋 로드
df = pd.read_csv("../data/train.csv", index_col="PassengerId")

# 데이터셋의 첫 5행 출력
print(df.head())

# 1등급 탑승객의 생존 비율 계산
first_class_survivors = df[df['Pclass'] == 1]['Survived'].sum()  # 1등급 승객 중 생존자 수
first_class_total = df[df['Pclass'] == 1]['Survived'].count()   # 1등급 승객 총 수

first_class_survival_rate = first_class_survivors / first_class_total  # 생존 비율 계산
print(f"1등급 탑승객 생존 비율: {first_class_survival_rate:.2%}")  # 비율 출력

# 2등급 탑승객의 생존 비율 계산
second_class_survivors = df[df['Pclass'] == 2]['Survived'].sum()  # 2등급 승객 중 생존자 수
second_class_total = df[df['Pclass'] == 2]['Survived'].count()   # 2등급 승객 총 수

second_class_survival_rate = second_class_survivors / second_class_total  # 생존 비율 계산
print(f"2등급 탑승객 생존 비율: {second_class_survival_rate:.2%}")  # 비율 출력

# 3등급 탑승객의 생존 비율 계산
third_class_survivors = df[df['Pclass'] == 3]['Survived'].sum()  # 3등급 승객 중 생존자 수
third_class_total = df[df['Pclass'] == 3]['Survived'].count()   # 3등급 승객 총 수

third_class_survival_rate = third_class_survivors / third_class_total  # 생존 비율 계산
print(f"3등급 탑승객 생존 비율: {third_class_survival_rate:.2%}")  # 비율 출력

             Survived  Pclass  \
PassengerId                     
1                   0       3   
2                   1       1   
3                   1       3   
4                   1       1   
5                   0       3   

                                                          Name     Sex   Age  \
PassengerId                                                                    
1                                      Braund, Mr. Owen Harris    male  22.0   
2            Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0   
3                                       Heikkinen, Miss. Laina  female  26.0   
4                 Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0   
5                                     Allen, Mr. William Henry    male  35.0   

             SibSp  Parch            Ticket     Fare Cabin Embarked  
PassengerId                                                          
1                1      0         A/5 21171   7.2500   NaN        S

높은 등급의 객실을 이용한 승객의 생존율이 더 높은 걸 볼 수 있다. 등급 객실이 높을 수록 생존율이 높을 걸 보니 부유층이 먼저 대피한 것으로 볼 수도 있다. 하지만 3등급 객실의 경우 타이타닉호의 구조를 살펴보면 구명정에서 가장 멀리 떨어져 있어 대피에 어려움이 상당했을 것으로도 추정할 수 있다.

In [107]:
# 1등급 객실에 탑승한 여성 데이터 필터링
first_class_women = df.loc[(df['Pclass'] == 1) & (df['Sex'] == 'female')]["Survived"]

# 생존율 계산
if len(first_class_women) > 0:
    rate_first_class_women = sum(first_class_women) / len(first_class_women)
else:
    rate_first_class_women = 0  # 생존율이 계산될 수 없는 경우

# 결과 출력
print("% of first class women who survived:", rate_first_class_women)

% of first class women who survived: 0.9680851063829787


1등급 객실의 여성 생존율은 무려 96%라고 나온다. 이를 보면 1등급 객실의 여성 생존율은 어마어마하게 높은 걸 볼 수 있다. 부유층과 사회적 약자라는 여성의 인식이 더해져 최우선적으로 대피했을 것이라는 추측을 해볼 수 있다.

In [116]:
# 1등급 객실에 탑승한 아이 데이터 필터링
first_class_children = df.loc[(df['Pclass'] == 1) & (df['Age'] <= 18)]["Survived"]

# 생존율 계산
if len(first_class_children) > 0:
    rate_first_class_children = sum(first_class_children) / len(first_class_children)
else:
    rate_first_class_children = 0  # 생존율이 계산될 수 없는 경우

# 결과 출력
print("% of first class children who survived:", rate_first_class_children)

% of first class children who survived: 0.875


1등급 객실에 탑승한 아이(18세 이하)의 생존율을 87.5%로 높게 나타났다. 1등급 객실의 여성과 마찬가지로 우선 대피하여 생존율이 높은 것으로 보인다.

In [110]:
# 1등급 객실에 탑승한 남성 데이터 필터링
first_class_men = df.loc[(df['Pclass'] == 1) & (df['Sex'] == 'male')]["Survived"]

# 생존율 계산
if len(first_class_men) > 0:
    rate_first_class_men = sum(first_class_men) / len(first_class_men)
else:
    rate_first_class_men = 0  # 생존율이 계산될 수 없는 경우

# 결과 출력
print("% of first class men who survived:", rate_first_class_men)

% of first class men who survived: 0.36885245901639346


1등급 객실에 탑승한 승객임에도 불구하고 남자의 생존율은 36퍼센트로 비교적 높지 않은 걸 볼 수 있다. 사회적 부유층이지만 1등급 객실의 여성,유아를 먼저 대피시키느라 생존율이 낮은 것으로 추측된다. 

In [118]:
# 3등급 객실에 탑승한 여성의 생존율 계산

# 총 여성 승객 수와 생존한 여성 승객 수
total_female_passengers = 165  # 총 여성 승객 수
survived_female_passengers = 76  # 생존한 여성 승객 수

# 생존율 계산
survival_rate = (survived_female_passengers / total_female_passengers) * 100

# 결과 출력
print(f"3등급 객실에 탄 여성의 생존율: {survival_rate:.2f}%")


3등급 객실에 탄 여성의 생존율: 46.06%


3등급 객실의 여성 생존율은 46%로 낮은 것처럼 보이지만, 3등급 객실 탑승객 생존율 24%에 비하면 비교적 높은 편이다. 3등급 객실 내에서도 여성 대피가 먼저 이루어진 것으로 보인다.

In [158]:
import pandas as pd

# CSV 파일 읽기
df = pd.read_csv("../data/train.csv", index_col="PassengerId")

# 여성 탑승객 수 구하기
total_women = df[df['Sex'] == 'female'].shape[0]

# 생존 여부가 기록된 여성 탑승객 수 구하기
survived_women = df[(df['Sex'] == 'female') & (df['Survived'].notna())].shape[0]

print("Total number of women passengers:", total_women)
print("Number of women with survival data:", survived_women)

Total number of women passengers: 314
Number of women with survival data: 314


트레인셋의 남녀 승객수를 알아보기 위해 작성해보았다.

In [160]:
import pandas as pd

# CSV 파일 읽기
df = pd.read_csv("../data/train.csv", index_col="PassengerId")

# 남성 승객 수 구하기
total_men = df[df['Sex'] == 'male'].shape[0]

print("Total number of male passengers:", total_men)

Total number of male passengers: 577


트레인셋 남녀 승객수를 비교해본 결과 남자가 훨씬 많이 탑승했다는 것을 알 수 있다. 밑에서 남녀 승객의 성별에 따른 생존율을 구해보기 위해 이 작업을 수행해보았다.

In [86]:
df = pd.read_csv("../data/train.csv", index_col="PassengerId") 
df.head()

women = df.loc[df.Sex == 'female']["Survived"]
rate_women = sum(women)/len(women)

print("% of women who survived:", rate_women)


% of women who survived: 0.7420382165605095


전체 여성에서 생존 여성의 비율은 74.2.퍼센트이다. 이것을 본다면 여성의 생존율은 매우 높은 것으로 보인다. 1912년 영국은 대부분의 다른 나라들과 마찬가지로 여성 참정권이 주어지지 않은 나라였다. 이 말은 여성을 남성과 동등한 위치로 보지 않았다는 뜻이고 남성은 여성을 약자로 보았기 때문에 약자를 보호해야한다는 의식으로 먼저 대피시켰기 때문에 여성의 생존율이 높았던 것으로 추측된다.


In [73]:
men = df.loc[df.Sex == 'male']["Survived"]
rate_men = sum(men)/len(men)

print("% of men who survived:", rate_men)

% of men who survived: 0.18890814558058924


전체 남성에서 생존 남성의 비율은 18.8퍼센트로 20퍼센트가 채되지 않는다. 이것을 보면 위의 여성의 경우와는 다르게 남성의 생존비율은 낮은 걸 볼 수 있다.

In [104]:
# 3. 0-10세 데이터 필터링 및 생존율 계산
age_00_10 = df.loc[(df['Age'] >= 0) & (df['Age'] <= 10)]["Survived"]
if len(age_00_10) > 0:
    rate_age_00_10 = sum(age_00_10) / len(age_00_10)
else:
    rate_age_00_10 = 0  # 생존율이 계산될 수 없는 경우

# 4. 결과 출력
print("% of 0-10 year olds who survived:", rate_age_00_10)


% of 0-10 year olds who survived: 0.59375


유아층(0-10세 아이들)의 생존율은 59%로 비교적 높은 걸 볼 수 있다. 재난상황에서 여성, 아이들을 먼저 대피시킨 것으로 보인다.

In [75]:
# 1. 데이터 로드
import pandas as pd
df = pd.read_csv("../data/train.csv", index_col="PassengerId")

# 2. 결측값 처리 (Age 컬럼)
df['Age'] = df['Age'].fillna(df['Age'].median())

# 3. 20-39세 데이터 필터링 및 생존율 계산
age_20_40 = df.loc[(df['Age'] >= 20) & (df['Age'] < 40)]["Survived"]
rate_age_20_40 = sum(age_20_40) / len(age_20_40)

# 4. 결과 출력
print("% of 20-40 year olds who survived:", rate_age_20_40)

% of 20-40 year olds who survived: 0.35815602836879434


20-39세의 생존 비율을 35.8% 정도로 비교적 낮다. 청년층이라고 불리는 나잇대의 사람들의 생존률은 비교적 낮은 걸 볼 수 있다.

In [23]:
# 3. 40-59세 데이터 필터링 및 생존율 계산
age_40_60 = df.loc[(df['Age'] >= 40) & (df['Age'] < 60)]["Survived"]
rate_age_40_60 = sum(age_40_60) / len(age_40_60)

# 4. 결과 출력
print("% of 40-60 year olds who survived:", rate_age_40_60)

% of 40-60 year olds who survived: 0.39416058394160586


청년층과 노년층의 사이에 있는 40세-59세의 사람들의 생존율은 39.4%로 20-39세의 생존율 보단 높았지만 생존율이 50%도 되지 않는 걸 보니 이 나이대의 사람들의 생존율 또한 낮은 걸 확인할 수 있다.

In [28]:
# 3. 60-69세 데이터 필터링 및 생존율 계산
age_60_70 = df.loc[(df['Age'] >= 60) & (df['Age'] < 70)]["Survived"]
rate_age_60_70 = sum(age_60_70) / len(age_60_70)

# 4. 결과 출력
print("% of 60-70 year olds who survived:", rate_age_60_70)

% of 60-70 year olds who survived: 0.3157894736842105


노년층 60-69세의 생존율은 31%로 낮게 나타났다. 청년층들보다 낮은 생존율을 보이고 있다. 